From cf910b5805eed715c49f7adadb46a97bc3b68e6a Mon Sep 17 00:00:00 2001
From: Bo Gan <ganb@vmware.com>
Date: Wed, 7 Dec 2022 02:05:47 +0000
Subject: [PATCH] Revert "PCI: Clear PCI_STATUS when setting up device"

This reverts commit 6cd514e58f12b211d638dbf6f791fa18d854f09c.

Write 0xffff0000 to command/status would cause the vmci PCI device to
reset. After reset, vmci PCI device has all BAR initialized to having
zero base. This would cause some other ACPI devices not being able to
reserve their IO port ranges, and kernel subsequently complains about
it. 

E.g.,

During probing:

  kernel: pci 0000:00:07.7: [15ad:0740] type 00 class 0x088000
  kernel: pci 0000:00:07.7: reg 0x10: [io  0x0000-0x003f]
  kernel: pci 0000:00:07.7: reg 0x14: [mem 0x00000000-0x0003ffff 64bit]

Later:

  kernel: pnp 00:00: disabling [io  0x0010-0x001f] because it overlaps 0000:00:07.7 BAR 0 [io  0x0000-0x003f]
  kernel: pnp 00:00: disabling [io  0x0024-0x0025] because it overlaps 0000:00:07.7 BAR 0 [io  0x0000-0x003f]
  kernel: pnp 00:00: disabling [io  0x0028-0x0029] because it overlaps 0000:00:07.7 BAR 0 [io  0x0000-0x003f]
  ...

This revert is a workaround. A proper solution is pending.

Signed-off-by: Bo Gan <ganb@vmware.com>
---
 drivers/pci/probe.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index c5286b027f00..bdcad5e0f057 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -1890,9 +1890,6 @@ int pci_setup_device(struct pci_dev *dev)
 
 	dev->broken_intx_masking = pci_intx_mask_broken(dev);
 
-	/* Clear errors left from system firmware */
-	pci_write_config_word(dev, PCI_STATUS, 0xffff);
-
 	switch (dev->hdr_type) {		    /* header type */
 	case PCI_HEADER_TYPE_NORMAL:		    /* standard header */
 		if (class == PCI_CLASS_BRIDGE_PCI)
-- 
2.25.1

